.github/workflows/smoke-test.yml

```diff
@@ -137,7 +137,7 @@ jobs:
     strategy:
       matrix:
         python-version:
-          - '3.9'
+          - '3.10'
```

Comment:
I'm reluctant to drop support for 3.9 due to a frozen version in a dev-dependency (ipython) that doesn't span 3.8-3.12. 

3.8 wasn't even EOL yet, but since it hits EOL in October, and it really simplified things, we dropped it.

I don't know what to suggest here, aside from installing from the requirements-dev.in on 3.9 or having separate pip-compile files.

Reply:
ipython 8.18 might be the last one that supports older versions?